internal/cloud/cloud.go

```diff
@@ -127,6 +145,45 @@ func Shell(ctx context.Context, w io.Writer, projectDir string, githubUsername s
 	return shell(username, vmHostname, projectDir, telemetryShellStartTime)
 }
 
+// Temporary function to create a vm and print vmHostname to be used by devbox extension
+func InitShell(ctx context.Context, w io.Writer, projectDir string, githubUsername string) error {
+	if err := ensureProjectDirIsNotSensitive(projectDir); err != nil {
+		return err
+	}
+
+	username, vmHostname := parseVMEnvVar()
+	// The flag for githubUsername overrides any env-var, since flags are a more
+	// explicit action compared to an env-var which could be latently present.
+	if githubUsername != "" {
+		username = githubUsername
+	}
+	if username == "" {
+		var err error
+		username, err = getGithubUsername()
+		if err != nil {
+			return err
+		}
+	}
+	debug.Log("username: %s", username)
+
+	// setup ssh config
+	sshCmd, err := SSHSetup(username)
+	if err != nil {
+		return err
+	}
+
+	// creating vm for user if it doesn't exist
+	vmHostname, err = ensureVMForUser(vmHostname, w, username, sshCmd)
+	if err != nil {
+		return err
+	}
+	debug.Log("vm_hostname: %s", vmHostname)
+	// printing vmHostname so that the output of devbox cloud init can be read by
+	// devbox extension
+	fmt.Fprintln(w, vmHostname)
```

Comment:
if you call `InitShell` from `cloud.Shell`, then instead of printing this here, you could return this value and print it in the `devbox cloud init` command implementation function...

Reply:
you may also have to return `telemetryShellStartTime`, if we're sharing the implementation...

Reply:
Can I just put `telemetryShellStartTime` in `Shell()` right before calling `InitVM`? 
That way I have 1 less item to return from InitVM.

Reply:
(Am on phone so can't quite see the code)

I think the problem is that getGithubUsername could fallback to a user prompt, and so we want the start time to begin after that...does that make sense?

Reply:
fair enough, moved the `telemetryShellStartTime` into `InitVM()`